packages/manager/modules/nasha/src/dashboard/dashboard.routing.js

```diff
@@ -0,0 +1,59 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('nasha.dashboard', {
+    url: '/:serviceName',
+    component: 'nashaDashboard',
+    redirectTo: 'nasha.dashboard.general-information',
+    resolve: {
+      breadcrumb: /* @ngInject */ (serviceName) => serviceName,
+      editName: /* @ngInject */ ($uibModal, nasha, goToDashboard) => () => {
+        const md = $uibModal.open({
```

Comment:
$uibModal is deprecated.

Reply:
The problem is that we want to change the name of the service by opening a modal, regardless of the child state we are on. I could put this edit modal into its own state and replicate this state on every child states. But for instance /{serviceName}/partitions/edit-name doesnt make any sens.

If I create a state like /{serviceName}/edit-name, once opened it will unload all the child states from the view

Unfortunately there is no other ways of doing it using the tools we have today

Plus, I was told that $uibModalInstance was deprecated in favour of the modal layout, not $uibModal. The $uibModal service is used a lot in the code

@radireddy any ideas ?